### PR TITLE
Fix report types selection not being overriden

### PR DESCRIPTION
**Pull request by Rieven**
### Changes

- At report types selection page, remove the previous selection from the report form fields, so that the new selection for report types is overriden.
- When all checkboxes are checked, the toggle-on class must be set other wise you have to click twice to un check all checked checkboxes.

### Issue link
https://github.com/minvws/nl-kat-coordination/issues/3434

Closes https://github.com/minvws/nl-kat-coordination/issues/3434

### QA notes

See issue link for for reproduction.

---

### Code Checklist

- [x] All the commits in this PR are properly PGP-signed and verified.
- [x] This PR only contains functionality relevant to the issue.
- [ ] I have written unit tests for the changes or fixes I made.
- [x] I have checked the documentation and made changes where necessary.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

- [x] Tickets have been created for newly discovered issues.
- [x] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [x] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [x] I have included comments in the code to elaborate on what is not self-evident from the code itself, including references to issues and discussions online, or implicit behavior of an interface.

---

## Checklist for code reviewers:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---

## Checklist for QA:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

File: rocky/assets/js/checkboxToggler.js [hunk 1]
```diff
@@ -1,4 +1,31 @@
...
+      ) {
+        if (checkboxes[i].checked) {
+          count_checked++;
+        }
+      }
+    }
+
+    if (count_checked == checkboxes.length) {
+      checkbox_target.classList.toggle("toggle-on");
+    }
```
Comment (ppvg) [inline rocky/assets/js/checkboxToggler.js:24]: The `toggle()` method will also remove the class if it's already present. Is that desired behaviour?

If not, you could consider using [`toggle()`'s `force` param](https://developer.mozilla.org/en-US/docs/Web/API/DOMTokenList/toggle#force), either to make sure it's only ever added:

```suggestion
    if (count_checked == checkboxes.length) {
      checkbox_target.classList.toggle("toggle-on", true);
    }
```

...or by using `count_checked == checkboxes.length` directly:

```suggestion
    checkbox_target.classList.toggle("toggle-on", count_checked == checkboxes.length);
```

Comment (Rieven) [reply rocky/assets/js/checkboxToggler.js:24]: Oeehhh nice parameters I can use indeed to get the desired results. thanks!

**Review by ammar92 — APPROVED** [review]

File: rocky/assets/js/checkboxToggler.js [hunk 2]
```diff
@@ -1,4 +1,32 @@
...
+      }
+    }
+
+    checkbox_target.classList.toggle(
+      "toggle-on",
+      count_checked == checkboxes.length,
+    );
+  }
+}
+SetToggleONWhenAllChecked();
```
Comment (underdarknl) [inline rocky/assets/js/checkboxToggler.js:28]: Execute this on dom load, not before.

File: rocky/assets/js/checkboxToggler.js [hunk 3]
```diff
@@ -1,4 +1,32 @@
...
+  checkbox_target = document.querySelector(".toggle-all[data-toggle-target]");
+  if (checkbox_target) {
+    var checkboxes = document.getElementsByName(
+      checkbox_target.dataset.toggleTarget,
+    );
+
+    for (var i = 0; i < checkboxes.length; i++) {
+      if (
+        checkboxes[i].tagName == "INPUT" &&
+        checkboxes[i].type == "checkbox"
```
Comment (underdarknl) [inline rocky/assets/js/checkboxToggler.js:14]: Can't these be handled in a direct dom query on line 7?

**Comment by underdarknl** [comment]
> * When all checkboxes are checked, the toggle-on class must be set other wise you have to click twice to un check all checked checkboxes.

This feels like a bug that we need to fix, instead of adding more code? When 'toggle all' is clicked, we should simply check if all available checkboxes have in fact been checked. We can do that by simple counting all checkboxes, and all checked checkboxes. Thats two simple queries against the dom, and avoids adding an extra class.

**Comment by Rieven** [comment]
> > * When all checkboxes are checked, the toggle-on class must be set other wise you have to click twice to un check all checked checkboxes.
> 
> This feels like a bug that we need to fix, instead of adding more code? When 'toggle all' is clicked, we should simply check if all available checkboxes have in fact been checked. We can do that by simple counting all checkboxes, and all checked checkboxes. Thats two simple queries against the dom, and avoids adding an extra class.

This is not a bug, the toggle is working as it supposed to when you are at the same page. But when all is checked and you go to the next page and you go back, then the toggle does not know on load that it needs to add `toggle-on` as default. It is only triggered when clicked.

**Comment by stephanie0x00** [comment]
## Checklist for QA:

- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
  - [x] I have managed to pass the onboarding flow
  - [x] Objects and Findings are created properly
  - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.
- [x] I checked the logs for errors and/or warnings and made issues where necessary

### What works:
The flow now works, the 'double select' is gone, and when you move forward and backward between the pages the selection holds and the plugins toggle along with it. 

### What doesn't work:
n/a

### Bug or feature?:
n/a

**Comment by underdarknl** [comment]
> > > * When all checkboxes are checked, the toggle-on class must be set other wise you have to click twice to un check all checked checkboxes.
> > 
> > 
> > This feels like a bug that we need to fix, instead of adding more code? When 'toggle all' is clicked, we should simply check if all available checkboxes have in fact been checked. We can do that by simple counting all checkboxes, and all checked checkboxes. Thats two simple queries against the dom, and avoids adding an extra class.
> 
> This is not a bug, the toggle is working as it supposed to when you are at the same page. But when all is checked and you go to the next page and you go back, then the toggle does not know on load that it needs to add `toggle-on` as suppose that the next click would uncheck all checkboxes. Because the toggle on is not set, it would first add it on the first click (checking all checkboxes that is already checked) and then the second click would uncheck all.

Sure, but adding this new code simply to add a  state that we could already ascertain during onLoad seems wasteful.
In OnLoad we can just check If all the available checkboxes are checked, and flip our internal state / set the class. Or am i missing something?

**Review by underdarknl — APPROVED** [review]

**Pull request merged**